tensorflow_datasets/scripts/refactoring/refactoring_dataset.py

```diff
@@ -0,0 +1,178 @@
+
+
+def copytree(src, dst):
+  """Copy files and folders recursively"""
+  for item in tf.io.gfile.listdir(src):
+    s = os.path.join(src, item)
+    d = os.path.join(dst, item)
+    if tf.io.gfile.isdir(s):
+      if tf.io.gfile.exists(d):
+        tf.io.gfile.rmtree(d)
+      shutil.copytree(s, d)
+    else:
+      if tf.io.gfile.exists(d):
+        tf.io.gfile.remove(d)
+      tf.io.gfile.copy(s, d)
+
+
+def make_init_file(dataset_type, dataset_name, dataset_path) -> None:
+  """Creates a new __init__.py. file"""
+  file_path = os.path.join(dataset_path, '__init__.py')
+  dataset_name_cls = naming.snake_to_camelcase(dataset_name)
+  data = {'dataset_type': dataset_type,
+          'dataset_name': dataset_name,
+          'dataset_name_cls': dataset_name_cls}
+
+  with tf.io.gfile.GFile(file_path, 'w') as f:
+    f.write(_INIT_FILE.format(**data))
+
+
+def copy_checksum_file(src_checksum_path, dest_path) -> None:
+  """Copy checksum.txt file"""
+  if tf.io.gfile.exists(src_checksum_path):
+    tf.io.gfile.copy(src_checksum_path,
+                     os.path.join(dest_path, posixpath.basename(src_checksum_path)),
+                     overwrite=True)
+
+
+def copy_make_data_file(src_fake_data_script_path, dest_path) -> None:
+  """Copy fake data genneration script file"""
+  if tf.io.gfile.exists(src_fake_data_script_path):
+    tf.io.gfile.copy(src_fake_data_script_path,
+                     os.path.join(dest_path, 'make_fake_data.py'),
+                     overwrite=True)
+
+
+def copy_fake_data_dir(src_fake_data_dir, dest_path) -> None:
+  """Copy fake data directory"""
+  if tf.io.gfile.exists(src_fake_data_dir):
+    copytree(src_fake_data_dir, os.path.join(dest_path, 'fake_data'))
+
+
+def copy_dataset_file(src_dataset_path, dest_path) -> None:
+  """Copy my_dataset.py file"""
+  tf.io.gfile.copy(src_dataset_path,
+                   os.path.join(dest_path, posixpath.basename(src_dataset_path)),
+                   overwrite=True)
+
+
+def copy_dataset_test_file(src_dataset_test_path, dest_path) -> None:
+  """Copy my_dataset_test.py file"""
+  tf.io.gfile.copy(src_dataset_test_path,
+                   os.path.join(dest_path, posixpath.basename(src_dataset_test_path)),
+                   overwrite=True)
+
+
+def refactor_dataset(datasets: Dict[str, list]) -> None:
+  """Refactor all the dataset"""
+  for dataset_type, dataset_names in datasets.items():
+    for dataset_name in dataset_names:
+      # Dataset files path
+      fake_data_dir = os.path.join(FAKE_DATA_DIR, dataset_name)
+      fake_data_script_py = f'{FAKE_DATA_SCRIPT_DIR}/{dataset_name}.py'
+      dataset_dir = os.path.join(TFDS_DIR, dataset_type)
+      checksum_txt = f'{URL_CHECKSUM_DIR}/{dataset_name}.txt'
+      dataset_py = f'{dataset_dir}/{dataset_name}.py'
+      dataset_test_py = f'{dataset_dir}/{dataset_name}_test.py'
+
+      # Newly creted dataset path
+      refactor_dataset_path = os.path.join(NEW_TFDS_DIR, dataset_type, dataset_name)
```

Comment:
New dataset should be in the same directory:

```
image/
  mnist.py
```

```
image/
  mnist/
    __init__.py
    mnist.py
    ...
```

Reply:
yeah, right now I made it for the purpose to separate refactored dataset. Will change this later on once we have a more robust script.

Reply:
Done